packages/client/lib/miner/miner.ts

```diff
@@ -257,7 +257,7 @@ export class Miner {
       },
       blockOpts: {
         cliqueSigner,
-        hardforkByBlockNumber: true,
```

Comment:
Do you know why it was on?

This caused a test to fail that involved building a `Fake Chain`, and arbitrarily assigning hardforks to block numbers.
```
    const customChainParams = {
      hardforks: [
        { name: 'chainstart', block: 0 },
        { name: 'berlin', block: 2 },
        { name: 'london', block: 3 },
      ],
    }
```


https://github.com/ethereumjs/ethereumjs-monorepo/blob/26e1890d800851bd9907a1978ff83e92594840a1/packages/client/test/miner/miner.spec.ts#L373

with `hardforkByBlockNumber` on, it would always revert back to `chainstart`, even with the   `common` option set to  `this.config.chainCommon`

Reply:
I think Ryan designed the test.

If there's a strong reason for having `hardforkByBlockNumber` on, I could instead look into changing `VM.buildBlock()` to ignore it if the `common` option is set?

Reply:
its on i think to make sure the correct block is produced according to hardfork at that current block number (could be a side chain/non-canonical chain and loading hardfork at that block number makes sense)

but how should your PR which doesn't look relevant to this, affect this logic, thats the first thing to understand why this spec test is failing.

Reply:
most likely the mockchain might not be build properly because of some validation failing, that the miner is building block from `0` where it seems like it should build post `3`. I will try this PR out tomorrow locally to confirm this hypo.

Reply:
Thanks I spent more time with it today and didn't get any further

Reply:
i think I have resolved the spec test, so turning this back on :+1: